### PR TITLE
Subcell NewtEul: Add subcell directory and DgInitialDataTci mutator

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Adds the subcell directory for Newtonian Euler
- Adds the TCI run on the initial data to figure out where the initial data is already bad

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by fmahebert — DISMISSED** [review]

File: src/Evolution/Systems/NewtonianEuler/Subcell/InitialDataTci.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VariablesTag.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/NewtonianEuler/Subcell/InitialDataTci.hpp]: unused

File: src/Evolution/Systems/NewtonianEuler/Subcell/InitialDataTci.hpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Tags/Inactive.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+/// \endcond
```
Comment (kidder) [inline src/Evolution/Systems/NewtonianEuler/Subcell/InitialDataTci.hpp:20]: forward declare Variables

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
I've rebased and squashed the changes. Thanks for the reviews :)

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**